tests/PhpSpreadsheetTests/Writer/Html/XssVulnerabilityTest.php

```diff
@@ -4,66 +4,42 @@
-
-        $spreadsheet->getActiveSheet()
-            ->getComment('A1')
-            ->setText($richText);
-
-        $filename = File::temporaryFilename();
-
-        $writer = IOFactory::createWriter($spreadsheet, 'Html');
-        $writer->save($filename);
-
-        $verify = file_get_contents($filename);
-        unlink($filename);
-        // Ensure that executable js has been stripped from the comments
-        self::assertStringContainsString($adjustedTextString, $verify);
-    }
-
     public static function providerXssRichText(): array
     {
         return [
-            'script tag' => ["Hello, I am trying to <script>alert('Hack');</script> your site"],
-            'javascript tag' => ["<a href='&#x2000;javascript:alert(1)'>CLICK</a>"],
-            'with unicode' => ['<a href="\\u0001java\\u0003script:alert(1)">CLICK<a>'],
-            'inline css' => ['<li style="list-style-image: url(javascript:alert(0))">'],
+            'basic text no problem' => ['Hello, I am safely viewing your site', 'Hello, I am safely viewing your site'],
+            'link no problem' => ["<a href='Visit Google'>Google is here</a>", "<a href='Visit Google'>Google is here</a>"],
+            'script tag' => ["Hello, I am trying to <script>alert('Hack');</script> your site", 'Hello, I am trying to  your site'],
+            'javascript tag no hex' => ["<a href='javascript:alert(1)'>CLICK</a>", "<a href='(1)'>CLICK</a>"],
+            'javascript tag' => ["<a href='&#x2000;javascript:alert(1)'>CLICK</a>", "<a href=' (1)'>CLICK</a>"],
+            'with unicode' => ['<a href="\\u0001java\\u0003script:alert(1)">CLICK<a>', '<a href="(1)">CLICK<a>'],
+            'inline css' => ['<li style="list-style-image: url(javascript:alert(0))">', '<li >'],
             'char value chevron' => ["\x3cscript src=http://www.example.com/malicious-code.js\x3e\x3c/script\x3e"],
+            'hexadecimal html' => ['<IMG SRC=&#106&#x61&#x76&#x61&#x73&#109&#x72&#x69&#x70&#x74&#x3A&#x61&#x6C&#x65&#x72&#x74&#x28&#x27&#x58&#x53&#x53&#x27&#x29>', '<IMG >'],
+            'iframe' => ['<iframe width="560" onclick="alert(\'xss\')" height="315" src="https://www.youtube.com/embed/whatever?rel=0&controls=0&showinfo=0" frameborder="0" allowfullscreen></iframe>', '&lt;iframe width="560"  height="315" src="https://www.youtube.com/embed/whatever?rel=0&controls=0&showinfo=0" frameborder="0" allowfullscreen&gt;&lt;/iframe&gt;'],
         ];
     }
 
     /**
      * @dataProvider providerXssRichText
+     *
+     * @param string $xssTextString
      */
-    public function testXssInComment(string $xssTextString): void
+    public function testXssInComment($xssTextString, ?string $expected = null): void
     {
         $spreadsheet = new Spreadsheet();
+        if ($expected === null) { // whole comment stripped away
+            $expected = ' <td class="column0 style0 s">XSS Test</td>';
+        } else {
+            $expected = '<div class="comment">' . $expected . '</div>';
+        }
```

Comment:
I know I gave you this code, but I have since thought of some improvements. Please replace this entire if statement with:
```php
        $startCell = '<td class="column0 style0 s">';
        $cellText = 'XSS Test';
        $endCell = $cellText . '</td>';
        if ($expected === null) { // whole comment stripped away
            $expected = $startCell . $endCell;
        } else {
            $expected = $startCell . '<a class="comment-indicator"></a><div class="comment">' . $expected . '</div>' . PHP_EOL . $endCell;
        }
```
I am having a bit of a problem with the review interface. In the statement below where you set the value to `'XSS Test'`, please set it to `$cellText` instead.

Reply:
done